### PR TITLE
2.6.0 Feature/glpi 92 compatibility

### LINKED ISSUE #640: GLPI 9.2 - $CFG_GLPI['use_mailing']
$CFG_GLPI['use_mailing'] changed with the new notification system. Needs an update to be comaptible with GLPI 9.2

### PR DESCRIPTION
**Pull request by btry**
# warning 

This PR breaks backward compatibility with GLPI 9.1 

fix #640

### CODE REVIEW

File: inc/entityconfig.class.php [hunk 1]
```diff
@@ -80,11 +80,11 @@ public function showFormForEntity(Entity $entity) {
       Dropdown::showFromArray('replace_helpdesk', $elements, array('value' => $this->fields['replace_helpdesk']));
       if ($this->fields['replace_helpdesk'] == self::CONFIG_PARENT) {
          $tid = self::getUsedConfig('replace_helpdesk', $ID);
-         echo "<font class='green'><br>";
```
Comment (orthagh) [inline inc/entityconfig.class.php]: OMG, this need a sacrifice !!! :smile:

**Comment by btry** [comment]
I merged drop of compatibility with GLPI 9.1.2+, rebased this branch and cleaned up remaining retrocompatibility code.

File: setup.php [hunk 2]
```diff
@@ -9,8 +9,6 @@
 define ('PLUGIN_FORMCREATOR_GLPI_MIN_VERSION', '9.2');
 // Maximum GLPI version, exclusive
 define ('PLUGIN_FORMCREATOR_GLPI_MAX_VERSION', '9.3');
-// Minimum version of PHP
-define('PLUGIN_FORMCREATOR_PHP_MIN_VERSION', '5.6.0');
```
Comment (btry) [inline setup.php]: GLPI already requires PHP 5.6, so this constant is no longer useful

File: setup.php [hunk 3]
```diff
@@ -35,27 +43,8 @@ function plugin_version_formcreator () {
...
-function plugin_formcreator_check_prerequisites () {
-   $success = true;
-   if (version_compare(GLPI_VERSION, PLUGIN_FORMCREATOR_GLPI_MIN_VERSION, 'lt')) {
-      echo 'This plugin requires GLPI >= ' . PLUGIN_FORMCREATOR_GLPI_MIN_VERSION . '<br>';
-      $success = false;
-   }
-   if (! function_exists('utf8_decode')) {
-      echo 'This plugin requires php-xml<br>';
-      $success = false;
-   }
```
Comment (btry) [inline setup.php]: GLPI 9.2 requires php-xml, so this check is redundant

File: setup.php [hunk 4]
```diff
@@ -35,27 +43,8 @@ function plugin_version_formcreator () {
  *
  * @return boolean
  */
-function plugin_formcreator_check_prerequisites () {
-   $success = true;
-   if (version_compare(GLPI_VERSION, PLUGIN_FORMCREATOR_GLPI_MIN_VERSION, 'lt')) {
-      echo 'This plugin requires GLPI >= ' . PLUGIN_FORMCREATOR_GLPI_MIN_VERSION . '<br>';
-      $success = false;
-   }
```
Comment (btry) [inline setup.php]: redundant with plugin_version_formcreator()

File: setup.php [hunk 5]
```diff
@@ -35,27 +43,8 @@ function plugin_version_formcreator () {
...
-
-   if (version_compare(PHP_VERSION, PLUGIN_FORMCREATOR_PHP_MIN_VERSION, 'lt')) {
-      if (method_exists('Plugin', 'messageIncompatible')) {
-         echo Plugin::messageIncompatible('core', PLUGIN_FORMCREATOR_GLPI_MIN_VERSION) . '<br/>';
-      } else {
-         echo 'This plugin requires PHP >=' . PLUGIN_FORMCREATOR_GLPI_MIN_VERSION . '<br>';
-      }
-      $success= false;
-   }
```
Comment (btry) [inline setup.php]: check PHP version no longer useful

File: setup.php [hunk 6]
```diff
@@ -64,14 +53,14 @@ function plugin_formcreator_check_prerequisites () {
  * @param string $verbose Set true to show all messages (false by default)
  * @return boolean
  */
-function plugin_formcreator_check_config($verbose=false) {
+function plugin_formcreator_check_config($verbose = false) {
```
Comment (btry) [inline setup.php]: CS

File: setup.php [hunk 7]
```diff
@@ -64,14 +53,14 @@ function plugin_formcreator_check_prerequisites () {
...
    return true;
 }
 
 /**
  * Initialize all classes and generic variables of the plugin
  */
-function plugin_init_formcreator () {
+function plugin_init_formcreator() {
```
Comment (btry) [inline setup.php]: CS

File: setup.php [hunk 8]
```diff
@@ -206,7 +195,7 @@ function plugin_init_formcreator () {
  * @param  String    $string  The string to encode
  * @return String             The encoded string
  */
-function plugin_formcreator_encode($string, $mode_legacy=true) {
+function plugin_formcreator_encode($string, $mode_legacy = true) {
```
Comment (btry) [inline setup.php]: CS

**Pull request merged**